KafkaIngester/src/main/java/io/deephaven/kafka/KafkaTools.java

```diff
@@ -195,20 +203,274 @@ else if (unionType0 == Schema.Type.NULL) {
+        return new KeyOrValueSpec.Simple(columnName, null);
+    }
+
+    public KeyOrValueSpec simpleSpec() {
+        return new KeyOrValueSpec.Simple(null, null);
+    }
+
+    public static Table consumeToTable(
+            @NotNull final Properties kafkaConsumerProperties,
+            @NotNull final String topic,
+            @NotNull final IntPredicate partitionFilter,
+            @NotNull final IntToLongFunction partitionToInitialOffset,
+            @NotNull final KeyOrValueSpec keySpec,
+            @NotNull final KeyOrValueSpec valueSpec) {
+        if (keySpec.dataFormat() == DataFormat.IGNORE && keySpec.dataFormat() == DataFormat.IGNORE) {
+            throw new IllegalArgumentException(
+                    "can't ignore both key and value: keySpec and valueSpec can't both be ignore specs");
+        }
+
+        final ColumnDefinition<?>[] commonColumns = new ColumnDefinition<?>[3];
+        getCommonCols(commonColumns, 0, kafkaConsumerProperties, partitionFilter == ALL_PARTITIONS);
+        final List<ColumnDefinition> columnDefinitions = new ArrayList<>();
+        int[] commonColumnIndices = new int[3];
+        int nextColumnIndex = 0;
+        for (int i = 0; i < 3; ++i) {
+            if (commonColumns[i] != null) {
+                commonColumnIndices[i] = nextColumnIndex++;
+                columnDefinitions.add(commonColumns[i]);
+            } else {
+                commonColumnIndices[i] = -1;
+            }
+        }
+
+        final MutableInt nextColumnIndexMut = new MutableInt(nextColumnIndex);
+        final KeyOrValueIngestData keyIngestData =
+                getIngestData(KeyOrValue.KEY, kafkaConsumerProperties, columnDefinitions, nextColumnIndexMut, keySpec);
+        final KeyOrValueIngestData valueIngestData =
+                getIngestData(KeyOrValue.VALUE, kafkaConsumerProperties, columnDefinitions, nextColumnIndexMut, valueSpec);
+
+        final TableDefinition tableDefinition = new TableDefinition(columnDefinitions);
+
+        final StreamPublisherImpl streamPublisher = new StreamPublisherImpl();
+        final StreamToTableAdapter streamToTableAdapter = new StreamToTableAdapter(tableDefinition, streamPublisher, LiveTableMonitor.DEFAULT);
+        streamPublisher.setChunkFactory(() -> streamToTableAdapter.makeChunksForDefinition(CHUNK_SIZE), streamToTableAdapter::chunkTypeForIndex);
+
+
+        final KeyOrValueProcessor keyProcessor = getProcessor(keySpec, tableDefinition, streamToTableAdapter, keyIngestData);
+        final KeyOrValueProcessor valueProcessor = getProcessor(valueSpec, tableDefinition, streamToTableAdapter, valueIngestData);
+
+        final ConsumerRecordToStreamPublisherAdapter adapter = KafkaStreamPublisher.make(streamPublisher,
+                commonColumnIndices[0],
+                commonColumnIndices[1],
+                commonColumnIndices[2],
+                keyProcessor, valueProcessor,
+                Function.identity(),
```

Comment:
should the transformation be part of the KeyIngestData, because doesn't JSON need to do the parsing here?